pygac/reader.py

```diff
@@ -78,120 +107,54 @@ def get_header_timestamp(self):
+                                dtype=counts.dtype)
+            channels[:, :, :2] = counts[:, :, :2]
+            channels[:, :, -2:] = counts[:, :, -2:]
+            channels[:, :, 2][switch == 1] = counts[:, :, 2][switch == 1]
+            channels[:, :, 3][switch == 0] = counts[:, :, 2][switch == 0]
             return channels
 
-    @abstractmethod
-    def _get_times(self):
-        """Specifies how to read scanline timestamps from GAC data.
-
-        Returns:
-            int: year
-            int: day of year
-            int: milliseconds since 00:00
-        """
-        raise NotImplementedError
-
-    def get_times(self):
-        """Read scanline timestamps and try to correct invalid values.
-
-        Note:
-            Also sets self.utcs and self.times!
-
-        Returns:
-            UTC timestamps
-        """
-        if self.utcs is None:
-            # Read timestamps
-            year, jday, msec = self._get_times()
-
-            # Correct invalid values
-            year, jday, msec = self.correct_times_median(year=year, jday=jday,
-                                                         msec=msec)
-            self.utcs = self.to_datetime64(year=year, jday=jday, msec=msec)
-            self.correct_times_thresh()
-
-            # Convert timestamps to datetime objects
-            self.times = self.to_datetime(self.utcs)
-
-        return self.utcs
-
-    @staticmethod
-    def to_datetime64(year, jday, msec):
-        """Convert timestamps to numpy.datetime64
-
-        Args:
-            year: Year
-            jday: Day of the year (1-based)
-            msec: Milliseconds since 00:00
-
-        Returns:
-            numpy.datetime64: Converted timestamps
-        """
-        return (((year - 1970).astype('datetime64[Y]')
-                + (jday - 1).astype('timedelta64[D]')).astype('datetime64[ms]')
-                + msec.astype('timedelta64[ms]'))
-
-    @staticmethod
-    def to_datetime(datetime64):
-        """Convert numpy.datetime64 to datetime.datetime
-
-        Args:
-            datetime64 (numpy.datetime64): Numpy timestamp to be converted.
-
-        Returns:
-            datetime.datetime: Converted timestamp
-        """
-        return datetime64.astype(datetime.datetime)
-
-    def lineno2msec(self, scan_line_number):
-        """Compute ideal scanline timestamp based on the scanline number.
-
-        Assumes a constant scanning frequency.
-
-        Args:
-            scan_line_number: Specifies the scanline number (1-based)
-
-        Returns:
-            Corresponding timestamps in milliseconds since 1970-01-01 00:00,
-            i.e. the first scanline has timestamp 0.
-        """
-        return (scan_line_number - 1) / self.scan_freq
-
     def compute_lonlat(self, utcs=None, clock_drift_adjust=True):
-        tle1, tle2 = self.get_tle_lines()
-
-        scan_points = np.arange(3.5, 2048, 5)
-
         if utcs is None:
             utcs = self.get_times()
```

Comment:
The origin of the timestamp mismatch between feature-lac and develop is here: `get_times()` returns a reference to `self.utcs`, so that decrementing utcs below (`utcs -= ...`) actually modifies `self.utcs`. I don't know whether this is intended or not. But I guess it is not, because each time `compute_lonlat()` is called, `self.utcs` changes its value. Using a local copy `utcs = self.get_times().copy()` resolves the timestamp mismatch. However, lat/lon is still different compared to the develop branch. The problem does not occur in the develop branch, because `compute_lonlat()` is never called with `utcs=None`.